bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -232,6 +231,7 @@ public void safeRun() {
                 sendWriteRequest(writeSet.get(i));
             }
         } finally {
+            hasRun = true;
```

Comment:
any reason why this is moved down to here?

Reply:
I thought it is good to set it after it actually finished running. Do you see any issues with it?

Reply:
there is a return at line 216? 

if the purpose is to setting this after it actually finished running, you should have pull all the logic in a try-finally block, no?

Reply:
yeah; I will move back the change.